### DISCUSSION TITLE
How change dafault IOS simulator?

### DISCUSSION THREAD

**Discussion by ZhorovEgor** [airsdk/Adobe-Runtime-Support#3298] [General]
Howdy! When I start debugging the programm, running the simulator using Xcode (Without using the built-in simulator or a direct connection to the iPhone via cable), my Ipad - 14.1 always starts, how do I change the standard value? For example, I need an iPhone 12.
![Screenshot_1](https://github.com/airsdk/Adobe-Runtime-Support/assets/63095086/f532e826-4007-4a76-8e33-91aadb8024c9)

**Comment by ajwfrost** [comment]
How do you launch it? If from Xcode, doesn't it just remember what you'd used previously?

If you're using ADT, you can use `-platform ios -device ios-simulator` and then specify which simulator to use via `-simulator "iPhone 12 Pro Max"` or similar. There's also an environment variable you could set for this, `AIR_IOS_SIMULATOR_DEVICE`. I think the possible names should match what you see in the menu options within the Simulator app itself..

And actually, it looks like we have an internal default which is just "iPhone 11". So, probably you could just `export AIR_IOS_SIMULATOR_DEVICE=iPhone 12 Pro Max` and then launch from ADT?

thanks

**Reply by ZhorovEgor** [reply]
Yes, I read that there is a way to do this using the commands that you prescribed, but I didn’t understand where to enter them?

**Reply by ZhorovEgor** [reply]
Howdy! @ajwfrost can you make an answer? I realy don't understand where i need put this command.

**Reply by ajwfrost** [reply]
Hi - this depends on how you're developing the app... are you using an IDE like IntelliJ IDEA, or Adobe Animate?

Just trying this on my mac:
- From Xcode, if I go on the menus Xcode -> Open Developer Tools -> Simulator, then the default simulator device that starts up is "iPhone 15" with iOS 17.5
- In the Simulator File menu, Open Simulator shows I have a fairly limited range of devices but e.g. under iOS 17.5 I can see "iPhone 15 Plus", "iPhone 15 Pro", "iPhone 15 Pro Max", "iPhone SE (3rd generation)", etc
- If I open one of these (Pro) and close the normal iPhone 15, then if I close the Simulator app and then re-open it, I will get the "iPhone 15 Pro" launching again
- This appears to be because in Simulator -> Settings, I have selected the option: "When Simulator starts boot the most recently used simulator"

So that can be used to set up a default simulator if that's how you're launching it.

But from AIR/ADT, if I've just compiled an IPA for a simulator, I can install and launch it using:
```
adt -installApp -platform ios -device ios-simulator -simulator "iPhone 15 Pro Max" -package Test.ipa
adt -launchApp  -platform ios -device ios-simulator -simulator "iPhone 15 Pro Max" -appid com.harman.air.Test
```
(and note I have the simulator SDK path set in my configuration file, otherwise you'd need to add the `-platformsdk` argument manually)

That launchApp command will then launch that specific simulator.

If you're using an IDE, and try to "run in simulator", does it start up any particular simulator? I'm not sure what behaviours they have, but for example I think Animate would just see if there are any simulators running, and if so would use that, rather than being able to specify a particular one?

Hope it helps..

**Reply by ZhorovEgor** [reply]
@ajwfrost this command 

> adt -installApp -platform ios -device ios-simulator -simulator "iPhone 15 Pro Max" -package Test.ipa
adt -launchApp  -platform ios -device ios-simulator -simulator "iPhone 15 Pro Max" -appid com.harman.air.Test


Do I need to compile on a standard Mac terminal?

**Reply by ajwfrost** [reply]
Those commands would run on a standard terminal. But, if you're using an IDE already, it may be possible to configure that to do the same thing. Depends on your IDE .... are you using anything like Animate/IDEA/VSCode here, or just the command line/terminal for building?!

**Reply by ZhorovEgor** [reply]
> Those commands would run on a standard terminal. But, if you're using an IDE already, it may be possible to configure that to do the same thing. Depends on your IDE .... are you using anything like Animate/IDEA/VSCode here, or just the command line/terminal for building?!

I'd worked in Adobe Animate.

**Reply by ajwfrost** [reply]
So from what I can tell in Animate, they end up always using the "default" simulator, but for me that appears to be "iPhone 15" rather than the actual last-used version. It's like they ignore that option and always use the version that Simulator would have used on its very first run...

Under the hood, they are calling ADT with the command lines per the above but of course without the option of choosing a simulator. We could add this to the list of feature requests for Animate of course........

It does work with that environment variable though. But unless there's a way to set system-wide environment variables in macOS (? I can check?) you would need to set it in a terminal window and then run Animate from the command line:
```
% export AIR_IOS_SIMULATOR_DEVICE="iPhone 15 Pro Max"
% echo $AIR_IOS_SIMULATOR_DEVICE
iPhone 15 Pro Max
% /Applications/Adobe\ Animate\ 2024/Adobe\ Animate\ 2024.app/Contents/MacOS/Adobe\ Animate\ 2024 
```

I guess we can add a configuration item into the AIR SDK settings (adt.cfg / AIR SDK Manager) to use instead of the environment variable, which might make it easier and more flexible....

**Reply by ZhorovEgor** [reply]
Thanks you!  But before using terminal need do this instruction:

Here is a step-by-step example of how to add the path to `adt` to the `PATH` variable.

Steps to add `adt` to the `PATH` variable

1. **Open the terminal**.
2. **Define the path to `adt`**:
Suppose `adt` is located in the directory `/Users/yourusername/AdobeAIRSDK/bin/adt'. You need to add `/Users/yourusername/AdobeAIRSDK/bin` to the `PATH` variable.

3. **Open the configuration file of your shell**:
   Depending on the shell you are using (bash, zsh, etc.), open the appropriate configuration file. For bash, this is '~/.bash_profile`, for zsh it is `~/.zshrc`. You can use the `nano` command to edit the file.

   ```sh
   nano ~/.bash_profile # for bash
   nano ~/.zshrc # for zsh
   ```

4. **Add the path to `adt` to the `PATH` variable**:
   Add the following line to the end of the file:

   ```sh
   export PATH=$PATH:/Users/yourusername/AdobeAIRSDK/bin
   ```

5. **Save the changes and close the editor**:
   In `nano`, press `Ctrl+O`, then `Enter` to save the file, and `Ctrl+X` to exit the editor.

6. **Reload the configuration file**:
   Run the following command in the terminal to apply the changes:

   ```sh
   source ~/.bash_profile # for bash
   source ~/.zshrc # for zsh
   ```

### Example of using `adt` after adding to `PATH`

Now you can use `adt` directly in the terminal. Here is an example of commands for creating and launching an app in iOS Simulator:

```sh
# Set the environment variable for the simulator
export AIR_IOS_SIMULATOR_DEVICE="iPad Retina"

# Application packaging
adt -package -target ipa-test-interpreter-simulator -storetype pkcs12 -keystore Certificates.p12 -storepass password WizDom.ipa WizDom-app.xml WizDom.swf -platformsdk /path/to/iPhoneSimulator.sdk

# Installing the app in the simulator
adt -installApp -platform ios -platformsdk /path/to/iPhoneSimulator.sdk -device ios-simulator -package WizDom.ipa

# Launching the app in the simulator
adt -LaunchApp -platform ios -platformsdk /path/to/iPhoneSimulator.sdk -device ios-simulator -appid your.app.id
``

Don't forget to replace `/path/to/iPhoneSimulator.sdk` with the correct path to the SDK and `your.app.id ` on your app's ID.

### A complete example for clarity

1. Open the configuration file:

   ```sh
   nano ~/.bash_profile
   ```

2. Add a line:

   ```sh
   export PATH=$PATH:/Users/yourusername/AdobeAIRSDK/bin
   ```

3. Save and close the file (`Ctrl+O`, `Enter`, `Ctrl+X').
4. Reload the configuration file:

   ```sh
   source ~/.bash_profile
   ```

Now 'adt` is accessible from the terminal and you can use its commands directly.
